CorsixTH/Lua/game_ui.lua

```diff
@@ -102,36 +103,36 @@ function GameUI:GameUI(app, local_hospital, map_editor)
   
   -- Set the scrolling keys.
   self.scroll_keys = {
-    [tostring(app.hotkeys["scroll_up"])]   = {x =   0, y = -10},
-    [tostring(app.hotkeys["scroll_down"])] = {x =  0, y =   10},
-    [tostring(app.hotkeys["scroll_left"])]  = {x =   -10, y =  0},
-    [tostring(app.hotkeys["scroll_right"])]  = {x = 10, y =   0},
+    [tostring(app.hotkeys["ingame_scroll_up"])]	= {x =   0, y = -10},
+    [tostring(app.hotkeys["ingame_scroll_down"])]	= {x =  0, y =   10},
+    [tostring(app.hotkeys["ingame_scroll_left"])]  = {x =   -10, y =  0},
+    [tostring(app.hotkeys["ingame_scroll_right"])]	= {x = 10, y =   0},
   }
 end
 
 function GameUI:setupGlobalKeyHandlers()
   UI.setupGlobalKeyHandlers(self)
 
-  self:addKeyHandler("escape", self, self.setEditRoom, false)
-  self:addKeyHandler("escape", self, self.showMenuBar)
-  self:addKeyHandler("z", self, self.keySpeedUp)
-  self:addKeyHandler("x", self, self.keyTransparent)
-  self:addKeyHandler({"shift", "a"}, self, self.toggleAdviser)
-  self:addKeyHandler({"ctrl", "d"}, self.app.world, self.app.world.dumpGameLog)
-  self:addKeyHandler({"ctrl", "t"}, self.app, self.app.dumpStrings)
-  self:addKeyHandler({"alt", "a"}, self, self.togglePlayAnnouncements)
-  self:addKeyHandler({"alt", "s"}, self, self.togglePlaySounds)
-  self:addKeyHandler({"alt", "m"}, self, self.togglePlayMusic)
+  self:addKeyHandler(self.app.hotkeys["global_window_close"], self, self.setEditRoom, false)
+  self:addKeyHandler(self.app.hotkeys["ingame_showmenubar"], self, self.showMenuBar)
+  self:addKeyHandler(self.app.hotkeys["ingame_gamespeed_speedup"], self, self.keySpeedUp)
+  self:addKeyHandler(self.app.hotkeys["ingame_setTransparent"], self, self.keyTransparent)
+  self:addKeyHandler(self.app.hotkeys["ingame_toggleAdvisor"], self, self.toggleAdviser)
+  self:addKeyHandler(self.app.hotkeys["ingame_poopLog"], self.app.world, self.app.world.dumpGameLog)
+  self:addKeyHandler(self.app.hotkeys["ingame_poopStrings"], self.app, self.app.dumpStrings)
+  self:addKeyHandler(self.app.hotkeys["ingame_toggleAnnouncements"], self, self.togglePlayAnnouncements)
+  self:addKeyHandler(self.app.hotkeys["ingame_toggleSounds"], self, self.togglePlaySounds)
+  self:addKeyHandler(self.app.hotkeys["ingame_toggleMusic"], self, self.togglePlayMusic)
   -- scroll to map position
   for i = 0, 9 do
     -- set camera view
-    self:addKeyHandler({"alt", tostring(i)}, self, self.setMapRecallPosition, i)
+    self:addKeyHandler({self.app.hotkeys["ingame_storePosition"], tostring(i)}, self, self.setMapRecallPosition, i)
```

Comment:
I haven't yet pulled this down to give it a try to visualise the change, but these image_storePosition/recallPosition are a list alt and ctrl modifiers for, 0 through 9. The change as it stands can only alter the default modifiers and leaving the numerical string as the 2nd element. I don't have an alternate suggestion.

Reply:
Fixing it's as easy as making a separate handler option for each. Should be done since the rest of the options expect / allow a table, and these ones can only be strings.

Reply:
Yes ingame_storePosition_0 etc is the solution, but if moving from the defaults to something else and a choice is made to rebind a subset. Say 0 to 4 is rebound to shift, QWERT, but 5-9 is not included and use the default.

Do we just leave the defaults for 6-0 or unbind them. I'm probably over thinking it and leaving defaults is fine and having them potentially disjoint is fine. As you could just want to remap a subset like the 6-0 to QWERT under the 1-5 row and then there is no need to explicitly set all of them.

Reply:
Yeah, I think that's fine.